### PR TITLE
[Enhance] Right align the log

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
The original log:
image

The aligned log:
image

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmengine/runner/log_processor.py [hunk 1]
```diff
@@ -140,20 +140,40 @@ def get_log_after_iter(self, runner, batch_idx: int,
         #  train: Epoch [5/10000] ... (divided by `max_iter`)
         #  val/test: Epoch [5/2000] ... (divided by length of dataloader)
         if self.by_epoch:
+            # Align the iteration log:
+            # Epoch(train)  [9][010/270]
```
Comment (zhouzaida) [inline mmengine/runner/log_processor.py:144]:
```suggestion
            # Epoch(train)  [9][ 10/270]
```

File: mmengine/runner/log_processor.py [hunk 2]
```diff
@@ -140,20 +140,40 @@ def get_log_after_iter(self, runner, batch_idx: int,
...
+            # ...               ||| |||
+            # Epoch(train)  [9][100/270]
+            max_dataloader_len = len(current_loop.dataloader)
+            max_str_len = len(str(max_dataloader_len))
```
Comment (zhouzaida) [inline mmengine/runner/log_processor.py:148]:
```suggestion
            dataloader_len = len(current_loop.dataloader)
            max_str_len = len(str(dataloader_len))
```

Comment (zhouzaida) [reply mmengine/runner/log_processor.py:148]: The length of dataloader is constant so the prefix `max_` should not be added.

File: mmengine/runner/log_processor.py [hunk 3]
```diff
@@ -140,20 +140,40 @@ def get_log_after_iter(self, runner, batch_idx: int,
...
+            cur_iter_str = str(cur_iter).rjust(max_str_len)
+
             if mode in ['train', 'val']:
+                # Right Align the epoch log:
+                # Epoch(train)   [9][100/270]
+                # ...             ||
+                # Epoch(train) [100][100/270]
                 cur_epoch = self._get_epoch(runner, mode)
+                max_epochs = runner.max_epochs
+                max_str_len = len(str(max_epochs)) + 3
```
Comment (zhouzaida) [inline mmengine/runner/log_processor.py:158]: The `max_str_len` in line 148 means the length of dataloader but `max_epochs + 3` here, which will confuse users.

Comment (HAOCHENYE) [reply mmengine/runner/log_processor.py:158]: 3 means the three characters: "[", "]",  and " " occupied in " [{max_epochs}]"

File: mmengine/runner/log_processor.py [hunk 4]
```diff
@@ -140,20 +140,40 @@ def get_log_after_iter(self, runner, batch_idx: int,
...
+            cur_iter_str = str(cur_iter).rjust(len(str(dataloader_len)))
+
             if mode in ['train', 'val']:
+                # Right Align the epoch log:
+                # Epoch(train)   [9][100/270]
+                # ...             ||
+                # Epoch(train) [100][100/270]
                 cur_epoch = self._get_epoch(runner, mode)
+                max_epochs = runner.max_epochs
+                # 3 means the three characters: "[", "]",  and " " occupied in
```
Comment (zhouzaida) [inline mmengine/runner/log_processor.py:157]:
```suggestion
                # 3 means the three characters: "[", "]", and " " occupied in
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**